sycl/test-e2e/Matrix/element_wise_all_ops_half_impl.hpp

```diff
@@ -41,11 +41,8 @@ void matrix_verify_add(queue q, big_matrix<T, M, N> &A, nd_range<2> &r,
 
            joint_matrix_fill(sg, sub_a, 5);
 
-           auto wi_slice_a =
-               sycl::ext::intel::experimental::matrix::get_wi_data(sg, sub_a);
-           for (int i = 0; i < wi_slice_a.length(); i++) {
-             wi_slice_a[i] = wi_slice_a[i] + static_cast<half>(2);
-           }
+           joint_matrix_apply(sg, sub_a,
```

Comment:
i saw some sycl testcases are using detail's thing:
llvm/sycl/test/regression/async_work_group_copy.cpp
could we use detail::get_wi_data in these elemwise testcase? @dkhaldi @YuriPlyakhin

Reply:
In my opinion, no, we can not use detail namespace in end-to-end tests (sycl/test-e2e). We should use only public/documented API in end-to-end tests.

Reply:
if so, we can delete them in sycl/test-e2e. maybe we can keep them in internal testing?

Reply:
In the current format, this test is testing each operation, as internally get_wi_data and each operation on it will be tested. So I don't understand the proposed change to this test by @YuriPlyakhin